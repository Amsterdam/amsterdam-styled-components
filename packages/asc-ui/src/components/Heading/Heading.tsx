import React from 'react'
import HeadingStyle from './HeadingStyle'

const Heading: React.FC<
  {
    as?: keyof JSX.IntrinsicElements | React.ComponentType<any>
  } & React.HTMLAttributes<HTMLHeadingElement>
> = ({ children, className, as, ...otherProps }) => (
  <HeadingStyle className={className} as={as} {...otherProps}>
    {children}
  </HeadingStyle>
)

Heading.defaultProps = {
  as: 'h1',
}

export default Heading
