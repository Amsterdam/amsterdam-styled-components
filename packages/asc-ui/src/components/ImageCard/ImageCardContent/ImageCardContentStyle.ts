import styled from '@datapunt/asc-core'
import { color, breakpoint } from '../../../utils'

export default styled.div`
  padding: 8px 14px;
  background-color: ${color('tint', 'level1')};
  position: absolute;
  bottom: 56px;
  left: 0;
  max-width: calc(100% - 44px);
  word-break: break-word;

  @media screen and ${breakpoint('min-width', 'tabletM')} {
    padding: 16px;
  }
`
