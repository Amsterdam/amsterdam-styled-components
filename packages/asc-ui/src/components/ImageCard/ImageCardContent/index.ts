export { default as ImageCardContentStyle } from './ImageCardContentStyle'
export { default as ImageCardContent } from './ImageCardContent'
