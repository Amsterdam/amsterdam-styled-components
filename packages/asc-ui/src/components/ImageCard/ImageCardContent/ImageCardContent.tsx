import React from 'react'
import ImageCardContentStyle, { Props } from './ImageCardContentStyle'

const ImageCardContent: React.FC<Props & React.HTMLAttributes<HTMLElement>> = ({
  children,
  ...otherProps
}) => <ImageCardContentStyle {...otherProps}>{children}</ImageCardContentStyle>

export default ImageCardContent
