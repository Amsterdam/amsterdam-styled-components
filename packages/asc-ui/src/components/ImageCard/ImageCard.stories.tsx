import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs'
import ImageCard from './ImageCard'
import { ImageCardContent } from './ImageCardContent'
import Heading from '../Heading'
import Paragraph from '../Paragraph'

storiesOf('Atoms/ImageCard', module)
  .addDecorator(withKnobs)
  .add('default state', () => (
    <div style={{ maxWidth: '600px' }}>
      <ImageCard backgroundImage="http://lorempixel.com/output/food-q-c-640-480-3.jpg">
        <ImageCardContent>
          <Heading $as="h4" styleAs="h2">
            Jeugdwerkloosheid Amsterdam daalt naar 6,2%
          </Heading>
        </ImageCardContent>
      </ImageCard>
    </div>
  ))
  .add('loading state', () => (
    <div style={{ maxWidth: '600px' }}>
      <ImageCard
        isLoading
        backgroundImage="http://lorempixel.com/output/food-q-c-640-480-3.jpg"
      >
        <ImageCardContent>
          <Heading $as="h4" styleAs="h2">
            Jeugdwerkloosheid Amsterdam daalt naar 6,2%
          </Heading>
          <Paragraph gutterBottom={0}>
            Amsterdamse jongeren even vaak werkloos als gemiddeld in Nederland
          </Paragraph>
        </ImageCardContent>
      </ImageCard>
    </div>
  ))
