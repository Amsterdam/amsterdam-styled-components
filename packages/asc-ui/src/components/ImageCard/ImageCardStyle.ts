import styled, { css } from '@datapunt/asc-core'
import { perceivedLoading } from '../../utils/themeUtils'
import { ImageCardContentStyle } from './ImageCardContent'

export type Props = {
  backgroundImage: string
  isLoading?: boolean
  animateLoading?: boolean
}

export type WrapperProps = {
  margin?: number
}

export const ImageCardWrapperStyle = styled.div<WrapperProps>`
  ${({ margin }) =>
    margin &&
    css`
      margin: ${margin}px;
    `}
`

export default styled.div<Props>`
  padding-top: 100%;
  background-size: cover;
  background-repeat: no-repeat;
  position: relative;
  width: 100%;
  overflow: hidden;
  ${({ isLoading, theme, animateLoading, backgroundImage }) =>
    isLoading
      ? css`
          ${perceivedLoading(theme, animateLoading)}
          ${ImageCardContentStyle} {
            height: 100px;
            width: 100%;

            & > * {
              display: none;
            }

            &::before {
              content: '';
              display: block;
              width: calc(100% - 30px);
              height: 30px;
              ${perceivedLoading(theme, animateLoading)}
            }
          }
        `
      : css`
          background-image: url(${backgroundImage});
        `}

  &::before {
    content: '';
    width: 0;
    height: 0;
    border-width: 22px;
    border-color: transparent;
    position: absolute;
    border-style: solid;
    right: 0;
    bottom: 0;
    border-bottom-color: white;
    border-right-color: white;
  }
`
