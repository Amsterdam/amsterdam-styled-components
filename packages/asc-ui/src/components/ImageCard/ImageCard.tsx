import { FunctionComponent, HTMLAttributes } from 'react'
import ImageCardStyle, {
  ImageCardWrapperStyle,
  WrapperProps,
  Props,
} from './ImageCardStyle'

const ImageCard: FunctionComponent<
  Props & WrapperProps & HTMLAttributes<HTMLElement>
> = ({ children, backgroundImage, margin, ...otherProps }) => (
  <ImageCardWrapperStyle margin={margin}>
    <ImageCardStyle backgroundImage={backgroundImage} {...otherProps}>
      {children}
    </ImageCardStyle>
  </ImageCardWrapperStyle>
)

export default ImageCard
