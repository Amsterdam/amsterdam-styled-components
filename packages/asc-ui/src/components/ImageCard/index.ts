import { ImageCardContent, ImageCardContentStyle } from './ImageCardContent'
import ImageCardStyle from './ImageCardStyle'

export { default } from './ImageCard'

export { ImageCardContent }

export const ImageCardStyles = {
  ImageCardContentStyle,
  ImageCardStyle,
}
