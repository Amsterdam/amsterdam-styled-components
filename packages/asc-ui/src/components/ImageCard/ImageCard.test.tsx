import { render } from '@testing-library/react'
import ImageCard from './ImageCard'
import Heading from '../Heading'
import ImageCardContent from './ImageCardContent/ImageCardContent'

describe('ImageCard', () => {
  it('should render', () => {
    const { container } = render(
      <ImageCard backgroundImage="https://www.example.com">
        <ImageCardContent>
          <Heading forwardedAs="h4" styleAs="h2">
            Jeugdwerkloosheid Amsterdam daalt naar 6,2%
          </Heading>
        </ImageCardContent>
      </ImageCard>,
    )

    expect(container.firstChild).toBeDefined()
  })
})
