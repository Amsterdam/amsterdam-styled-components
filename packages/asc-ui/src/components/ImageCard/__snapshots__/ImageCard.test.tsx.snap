// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCard should render 1`] = `
.c3 {
  margin: 0;
  color: #000000;
  font-weight: 700;
  font-size: 20px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 28px;
  margin-bottom: 8px;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c2 {
  margin-top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  padding: 8px 14px;
  background-color: #ffffff;
  position: absolute;
  bottom: 56px;
  left: 0;
  max-width: calc(100% - 44px);
  word-break: break-word;
}

.c0 {
  padding-top: 100%;
  background-size: cover;
  background-repeat: no-repeat;
  position: relative;
  width: 100%;
  overflow: hidden;
  background-image: url(https://www.example.com);
}

.c0::before {
  content: '';
  width: 0;
  height: 0;
  border-width: 22px;
  border-color: transparent;
  position: absolute;
  border-style: solid;
  right: 0;
  bottom: 0;
  border-bottom-color: white;
  border-right-color: white;
}

@media screen and (min-width:1024px) {
  .c3 {
    font-size: 24px;
    line-height: 30px;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    padding: 16px;
  }
}

<div
  class=""
>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h4
        class="c2 c3"
      >
        Jeugdwerkloosheid Amsterdam daalt naar 6,2%
      </h4>
    </div>
  </div>
</div>
`;
